skiplang/prelude/src/stdlib/collections/persistent/SortedMap.sk

```diff
@@ -326,14 +326,17 @@ base class .SortedMap<+K: Orderable, +V> uses  //  extends KeyedSequence<K, V>
     f: (V2, V2) -> V2,
   ): SortedMap<K2, V2>
   | Nil() -> SortedMap::singleton(key, value)
-  | Node _ ->
+  | n @ Node _ ->
     l: SortedMap<K2, V2> = this.left;
     r: SortedMap<K2, V2> = this.right;
     k: K2 = this.key;
     v: V2 = this.value;
     compare(key, this.key) match {
     | LT() -> SortedMap::balance(k, v, l.setWith(key, value, f), r)
-    | EQ() -> SortedMap::node(k, f(v, value), l, r)
+    | EQ() ->
+      newV = f(v, value);
+      node: Node<K2, V2> = n;
+      node with {value => newV}
```

Comment:
For my understanding, this optimization only saves the re-calculation of the new node's size and height; in particular, it is a few loads and arithmetic instructions, but there is no memory reuse, right?

Reply:
Unless the compiler becomes more clever, there is no memory reuse indeed.
I don't know how the `with` is compiled but it seems it could be more easily optimized by the compiler.

The change made more sense when it was followed by the `phys_eq` optimization. Here, it just makes it clearer that the tree's shape is not changed, only the value is updated.